app/controllers/users_controller.rb

```diff
@@ -0,0 +1,48 @@
+class UsersController < ApplicationController
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    user = User.new just_invited_user
+    if user.save
+      flash[:notice] = "Success!"
+      UserMailer.invitation(user.id).deliver_now
+    else
+      flash[:notice] = "You must be missing something."
+    end
+    redirect_to new_user_path
+  end
+
+  def edit
+    @user = User.find params[:id]
+    confirmation_hash = params[:token]
+    if confirmation_hash != @user.confirmation_hash
+      flash[:notice] = "It looks like you're lost - look at some art"
+      redirect_to root_path
+    end
+  end
+
+  def update
+    user = User.find params[:id]
+    user.update user_params
+    flash[:notice] = "Success! Welcome to the Baltimore Public Art Commons Team!"
+    redirect_to root_path
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit :email,
+                                 :role,
+                                 :passowrd,
+                                 :password_confirmation
+  end
+
+  def just_invited_user
```

Comment:
Should we name this something else? `just_invited_user` is a little misleading since the method doesn't return a user.